aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -285,6 +285,9 @@ Tensor linalg_pinv(
     const optional<Tensor>& atol_opt,
     const optional<Tensor>& rtol_opt,
     bool hermitian) {
+  // FIXME: Whenever we have a nice lstsq, we should dispatch this function to simply be
+  // `torch.lstsq(A, torch.eye(A.shape[-1]), atol=atol, rtol=rtol)`
+  // with a driver that supports singular inputs
```

Comment:
gelsy and all SVD-based drivers do support singular inputs but only for the CPU.

Reply:
For CUDA only `driver='gels'` is implemented.

Reply:
What I mean there is that it'd be good to move the code we have here into `lstsq` to support "manually" the SVD modes on GPU as well. This way we could have this functionality here and in `lstsq`.

Reply:
We can still dispatch it for the CPU if needed.